### PR TITLE
Update ImGui

### PR DESCRIPTION
**Pull request by wagrenier**
* Uses new Table API
* Fixes a mishandling of EndChild
* Updates imgui to Dock Branch v1.90

### CODE REVIEW

File: vutrace.cpp [hunk 1]
```diff
@@ -444,16 +453,15 @@ void memory_window(AppState &app)
...
 		}
-		
+
 		ImGui::PopStyleColor();
 		ImGui::PopStyleVar();
 		ImGui::PopStyleVar();
-		
-		ImGui::EndChild();
 	}
+    ImGui::EndChild();
```
Comment (wagrenier) [inline vutrace.cpp:464]: Github is making the tab misaligned, all is good in actual code

**Comment by chaoticgd** [comment]
Few things:
- I don't understand why cimgui was added. That's a wrapper for C projects, it's completely irrelevant here. Actually, it looks like there was an entry for it in the .gitmodules file previously, but I have no idea why that was there.
- I'd prefer if a release version of imgui was used (see the v1.90.4-docking branch) rather than a random commit between releases.
- There's a lot of mixed tabs and spaces and other whitespace changes, I'd prefer if it was all tabs. Otherwise anyone with their tab width not set to 4 columns will have to deal with broken indentation.
- I don't understand why the path of imgui in .gitmodules was removed.

**Comment by chaoticgd** [comment]
Also just FYI the reason for the includes to files in examples/ was because that's where the backends previously were, they were moved a couple years ago because it was misleading.

**Comment by wagrenier** [comment]
I'll make those changes. For cimgui, it just auto downloaded itself when I initialized the modules. 

I can create a style file for your project so that your preferred style is always enforced; this would include the tabs rules.

**Comment by chaoticgd** [comment]
If you can come up with a clang-format file that matches. I've tried multiple times, but I've never come up with one that I've been happy with myself. Always gets something wrong. Alternatively a .editorconfig file would also work.

**Comment by wagrenier** [comment]
I'll take care of the tabs in this PR and make a style file later on. Regarding the imgui submodules, for me the path is the same:
image
I think the default value is the path you gave hence why it does not show up?

**Comment by wagrenier** [comment]
I copy pasted back your original function with the blank spaces directly from what was on your main branch. On github it shows that it is different for some reason

**Comment by chaoticgd** [comment]
It's still using spaces instead of tabs, that's probably why it's showing up as different. It might be your editor doing it automatically.

**Comment by chaoticgd** [comment]
A good tip is to always check `git diff HEAD` before committing (but after `git add`ing), leads to less of this faffing.

**Comment by chaoticgd** [comment]
I'm just gonna close this and do it myself.

Don't bother with making a .clang-format file. The problem isn't the small formatting mistakes, it's the large diffs that touch lots of unrelated stuff.

**Comment by wagrenier** [comment]
Just add me in the contributors for the patch notes please. Some work was still implicated on fixing tables.

I'll make my next PR to update the rest of the app to the new APIs

**Comment by wagrenier** [comment]
I just noticed that it does not include the Table API, I'll make a pr to update all of that. Should be up this weekend

**Pull request closed without merging**